tests/connection_state_test.c

```diff
@@ -1748,12 +1748,12 @@ static int s_check_packets_received_order(
     uint16_t packet_id_2,
     uint16_t packet_id_3) {
     ASSERT_SUCCESS(mqtt_mock_server_decode_packets(handler));
-    int packet_idx_1 = 0;
-    int packet_idx_2 = 0;
-    int packet_idx_3 = 0;
-    ASSERT_NOT_NULL(mqtt_mock_server_find_decoded_packet_by_ID(handler, search_start_idx, packet_id_1, &packet_idx_1));
-    ASSERT_NOT_NULL(mqtt_mock_server_find_decoded_packet_by_ID(handler, search_start_idx, packet_id_2, &packet_idx_2));
-    ASSERT_NOT_NULL(mqtt_mock_server_find_decoded_packet_by_ID(handler, search_start_idx, packet_id_3, &packet_idx_3));
+    size_t packet_idx_1 = 0;
+    size_t packet_idx_2 = 0;
+    size_t packet_idx_3 = 0;
+    ASSERT_NOT_NULL(mqtt_mock_server_find_decoded_packet_by_id(handler, search_start_idx, packet_id_1, &packet_idx_1));
```

Comment:
Not really related, maybe we can rename the handler/test_channel_handler in this file to something like `mock_server`?